demo/src/init.rs

```diff
@@ -53,15 +52,15 @@ pub fn rendering_init(
     sdl2_window: &sdl2::video::Window,
     asset_source: AssetSource,
 ) -> RafxResult<()> {
-    resources.insert(SpriteRenderNodeSet::default());
-    resources.insert(MeshRenderNodeSet::default());
-    resources.insert(TileLayerRenderNodeSet::default());
     resources.insert(StaticVisibilityNodeSet::default());
     resources.insert(DynamicVisibilityNodeSet::default());
-    resources.insert(DebugDraw3DResource::new());
-    resources.insert(TextResource::new());
     resources.insert(ViewportsResource::default());
-    resources.insert(TileLayerResource::default());
+
+    mesh::legion_init(resources);
```

Comment:
I wonder if it would be possible for a user to provide a closure that takes a resource and inserts it into the resources. (We can investigate that some other time, this change is already huge.)